daffodil-core/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/ElementCombinator1.scala

```diff
@@ -201,7 +202,7 @@ abstract class ElementParserBase(
       // if we should do limited validation via CheckConstraints
       val shouldValidate = erd.isSimpleType &&
         pstate.dataProc.isDefined &&
-        pstate.dataProc.value.validationMode == ValidationMode.Limited
+        pstate.dataProc.value.validator == DaffodilLimitedValidator
```

Comment:
This might want to be `eq` to ensure we're doing reference equality, which should be faster.